exp/controllers/azuremachinepool_controller.go

```diff
@@ -255,36 +239,40 @@ func (r *AzureMachinePoolReconciler) reconcileNormal(ctx context.Context, machin
 
 	ams := newAzureMachinePoolService(machinePoolScope, clusterScope)
 
-	// Get or create the virtual machine.
-	vmss, err := ams.CreateOrUpdate(ctx)
+	err := ams.Reconcile(ctx)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// Make sure Spec.ProviderID is always set.
-	machinePoolScope.AzureMachinePool.Spec.ProviderID = fmt.Sprintf("azure:///%s", vmss.ID)
-	providerIDList := make([]string, len(vmss.Instances))
-	var readyCount int32
-	for i, vm := range vmss.Instances {
-		providerIDList[i] = fmt.Sprintf("azure:///%s", vm.ID)
-		if vm.State == infrav1.VMStateSucceeded {
-			readyCount++
-		}
-	}
-	machinePoolScope.AzureMachinePool.Spec.ProviderIDList = providerIDList
-	machinePoolScope.AzureMachinePool.Status.ProvisioningState = &vmss.State
-	machinePoolScope.AzureMachinePool.Status.Replicas = int32(len(providerIDList))
-	machinePoolScope.SetAnnotation("cluster-api-provider-azure", "true")
-
-	switch vmss.State {
+	switch machinePoolScope.ProvisioningState() {
 	case infrav1.VMStateSucceeded:
-		machinePoolScope.Info("Machine Pool is running", "id", *machinePoolScope.GetID())
+		machinePoolScope.V(2).Info("Scale Set is running", "id", machinePoolScope.GetProviderID())
```

Comment:
should we also add conditions here for machinepools? not necessarily in this PR, but I can create an issue if its makes sense

Reply:
yes definitely. Related to https://github.com/kubernetes-sigs/cluster-api/issues/3497